lib/omise/logger.rb

```diff
@@ -0,0 +1,53 @@
+module Omise
+  class Logger
+    LABEL = "[Omise]".freeze
+
+    attr_reader :log
+
+    def initialize(log = nil)
+      @log = log
+    end
+
+    def log_request(request, args)
+      return unless log
+      return unless can_log_request?(request)
+
+      message =  with_line_break { "#{LABEL} Request: #{args[:method].to_s.upcase} #{args[:url]}" }
+      message << with_line_break { format_headers(request) }
+      message << with_line_break { format_payload(args[:payload]) } if args[:payload]
+
+      log.info(message)
+    end
+
+    def log_response(response)
+      return unless log
+
+      net_http = response.net_http_res
+
+      message =  with_line_break { "#{LABEL} Response: HTTP/#{net_http.http_version} #{net_http.code} #{net_http.message}" }
+      message << with_line_break { format_headers(net_http) }
+      message << with_line_break { response.body }
```

Comment:
We could use a StringIO here instead of using the with_line_break. And even possibly make a small wrapper like this:

```ruby
require "stringio"

class LogMessage
  def initialize
    @message = StringIO.new
  end

  def puts(str)
    @message.puts(str)
  end

  def to_s
    @message.rewind
    @message.read
  end
end

message = LogMessage.new
message.puts "hello"
message.puts "world"
message.to_s # => "hello\nworld\n"
```

And I think `log.info(message)` should call `to_s` or `inspect` somewhere down the line and everything should continue to work as expected. Otherwise just call `to_s` here directly.